met-web/src/components/engagement/form/EngagementWidgets/WhoIsListeningForm.tsx

```diff
@@ -2,31 +2,86 @@ import React, { useContext, useState } from 'react';
 import { Autocomplete, Grid, TextField } from '@mui/material';
 import { ActionContext } from '../ActionContext';
 import { MetLabel, PrimaryButton, SecondaryButton } from 'components/common';
-import { WidgetContact } from '../types';
+import { Contact } from 'models/contact';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { postWidgetItems } from 'services/widgetService';
+import { WidgetDrawerContext } from './WidgetDrawerContext';
+import ContantInfoPaper from './ContactInfoPaper';
+import { WidgetType } from 'models/widget';
+import WhoIsListeningCreate from './WhoIsListeningCreate';
 
 const WhoIsListeningForm = () => {
-    const { handleWidgetDrawerOpen } = useContext(ActionContext);
-    const [selectedContact, setSelectedContact] = useState<WidgetContact | null>(null);
+    const { savedEngagement } = useContext(ActionContext);
+    const { handleWidgetDrawerOpen, handleAddContactDrawerOpen, loadingContacts, contacts, widgets } =
+        useContext(WidgetDrawerContext);
+    const dispatch = useAppDispatch();
+    const [selectedContact, setSelectedContact] = useState<Contact | null>(null);
+    const [addedContacts, setAddedContacts] = useState<Contact[]>([]);
+    const [addingWidgetItems, setAddingWidgetItems] = useState(false);
 
-    const testContact = {
-        id: 0,
-        name: 'Mandy Ming',
-        title: 'inspector',
-        phoneNumber: '123-234-2345',
-        email: 'mandy@gov.bc.ca',
-        address: '123 Main St. Victoria, BC, V0V 1B1',
-        bio: 'Mandy is always happy to answer your questions and...',
+    const widgetId = widgets.filter((widget) => widget.widget_type_id === WidgetType.WhoIsListening)[0]?.id || null;
+
+    const addContact = () => {
+        if (!selectedContact || addedContacts.map((contact) => contact.id).includes(selectedContact.id)) {
```

Comment:
may be make this if into a variable for better readability..i am guessing this handles duplicate contact being added again

Reply:
good idea, done